pkg/ingress/ingresscontroller_test.go

```diff
@@ -0,0 +1,241 @@
+package ingress
+
+import (
+	"testing"
+
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func TestIngressPull(t *testing.T) {
+	testCases := []struct {
+		ingressName         string
+		ingressNamespace    string
+		expectedError       bool
+		addToRuntimeObjects bool
+		expectedErrorText   string
+	}{
+		{
+			expectedError:       false,
+			addToRuntimeObjects: true,
+			expectedErrorText:   "",
+			ingressName:         "test",
+			ingressNamespace:    "test",
+		},
+		{
+			expectedError:       true,
+			addToRuntimeObjects: false,
+			expectedErrorText:   "ingresscontroller object test not found in namespace test",
+			ingressName:         "test",
+			ingressNamespace:    "test",
+		},
+		{
+			expectedError:       true,
+			addToRuntimeObjects: true,
+			expectedErrorText:   "ingresscontroller name cannot be empty",
+			ingressName:         "",
+			ingressNamespace:    "test",
+		},
+		{
+			expectedError:       true,
+			addToRuntimeObjects: true,
+			expectedErrorText:   "ingresscontroller namespace cannot be empty",
+			ingressName:         "test",
+			ingressNamespace:    "",
+		},
+	}
+
+	for _, testCase := range testCases {
+		var runtimeObjects []runtime.Object
+
+		var testSettings *clients.Settings
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, &operatorv1.IngressController{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testCase.ingressName,
+					Namespace: testCase.ingressNamespace,
+				},
+			})
+		}
+
+		testSettings = clients.GetTestClients(clients.TestClientParams{
+			K8sMockObjects: runtimeObjects,
+		})
+
+		builderResult, err := Pull(testSettings, testCase.ingressName, testCase.ingressNamespace)
+
+		if testCase.expectedError {
+			assert.NotNil(t, err)
+
+			if testCase.expectedErrorText != "" {
+				assert.Equal(t, testCase.expectedErrorText, err.Error())
+			}
+		} else {
+			assert.Nil(t, err)
+			assert.NotNil(t, builderResult)
+		}
+	}
+}
+
```

Comment:
Can you add tests for Get(),Exist(), Update - It's OK to have it commented.

Reply:
Done.